skimage/restoration/deconvolution.py

```diff
@@ -375,16 +376,27 @@ def richardson_lucy(image, psf, iterations=50, clip=True):
         convolve_method = fftconvolve
     else:
         convolve_method = convolve
-
     image = image.astype(np.float)
-    psf = psf.astype(np.float)
-    im_deconv = np.full(image.shape, 0.5)
+    psf = psf.astype(np.float)    
     psf_mirror = psf[::-1, ::-1]
-
-    for _ in range(iterations):
+    im_deconv = image
+    im_deconv_mid = np.zeros_like(image)
+    v = np.zeros_like(image)
+    for iter in range(iterations):
+        im_deconv_mid_prev = im_deconv_mid
         relative_blur = image / convolve_method(im_deconv, psf, 'same')
-        im_deconv *= convolve_method(relative_blur, psf_mirror, 'same')
-
+        im_deconv_mid = im_deconv * convolve_method(relative_blur, psf_mirror,
+                                                    'same')
+        v_prev = v
+        v = im_deconv_mid - im_deconv
+        np.maximum(v, 0, out=v)
+        if iter == 0:
+            alpha = 0
+        else:
+            alpha = (v_prev * v).sum() / ((v_prev * v_prev).sum() + 2e-15)
+            alpha = np.clip(alpha, 0, 1)
```

Comment:
clip also seems to provide an out parameter. In-place will make things faster for larger images.

Reply:
Thies alpha is just a single number

Reply:
haha, sorry.